internal/utils/utils.go

```diff
@@ -58,15 +74,15 @@ func GetSyncedAppsByStage(apps []argov1alpha1.Application, name string) []argov1
 
 	for _, app := range apps {
 		val, ok := app.Annotations[ProgressiveSyncSyncedAtStageKey]
-		if ok && val == name && app.Status.Sync.Status == argov1alpha1.SyncStatusCodeSynced && app.Status.Health.Status == health.HealthStatusHealthy {
+		if ok && val == name && app.Status.Sync.Status == argov1alpha1.SyncStatusCodeSynced {
```

Comment:
what was the rationale behind this removal?

Reply:
Because the function is called `GetSyncedAppsByStage` so it should get `synced` app in the given stage - meaning it has the annotation.
The old function was also looking for the health status and we didn't need that.